Nil pointer panic for untagged ECR image

**Description:**
Error log
```log
{"level":"info","ts":1585656805.7164814,"logger":"controller_api","msg":"Swagger version"}
{"level":"info","ts":1585656805.7165685,"logger":"controller_api","msg":"Deployment mode is not set in the swagger."}
{"level":"info","ts":1585656805.7165978,"logger":"controller_api","msg":"Endpoint name petstore.swagger.io"}
{"level":"info","ts":1585656805.716603,"logger":"controller_api","msg":"getting security instance","Request.Namespace":"default","Request.Name":"petstore-int"}
{"level":"info","ts":1585656805.7166085,"logger":"controller_api","msg":"API Level security is not defined"}
{"level":"info","ts":1585656805.7166672,"logger":"controller_api","msg":"Resource level security is not defiend"}
{"level":"info","ts":1585656805.7178092,"logger":"controller_api","msg":"Creating swagger configmap for mgw","name":"petstore-int-1-swagger-mgw","namespace":"default"}
{"level":"info","ts":1585656805.717841,"logger":"controller_api","msg":"use default security"}
{"level":"info","ts":1585656805.717852,"logger":"controller_api","msg":"default security exists in default namespace"}
{"level":"info","ts":1585656805.7178717,"logger":"registry","msg":"Setting registry type","registry-type":"AMAZON_ECR","repository":"610968236798.dkr.ecr.us-east-2.amazonaws.com/my-ecr-repo","image":"petstore-int","tag":"v1"}
{"level":"info","ts":1585656805.7178943,"logger":"controller_api","msg":"Docker credentials secret is not found","secret-name":"docker-registry-credentials","namespace":"wso2-system"}
{"level":"info","ts":1585656808.9762576,"logger":"controller_api","msg":"Found the image tag from the AWS ECR repository","RegistryId":"610968236798","RepositoryName":"my-ecr-repo","image":"petstore-int","tag":"v1"}
{"level":"info","ts":1585656808.9763052,"logger":"controller_api","msg":"Found the image tag from the AWS ECR repository","RegistryId":"610968236798","RepositoryName":"my-ecr-repo","image":"petstore-int","tag":"v1"}
{"level":"info","ts":1585656808.976312,"logger":"controller_api","msg":"Found the image tag from the AWS ECR repository","RegistryId":"610968236798","RepositoryName":"my-ecr-repo","image":"petstore-int","tag":"v1"}
E0331 12:13:28.976475       1 runtime.go:69] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
/home/renuka/gopath/pkg/mod/k8s.io/apimachinery@v0.0.0-20190221213512-86fb29eff628/pkg/util/runtime/runtime.go:76
/home/renuka/gopath/pkg/mod/k8s.io/apimachinery@v0.0.0-20190221213512-86fb29eff628/pkg/util/runtime/runtime.go:65
/home/renuka/gopath/pkg/mod/k8s.io/apimachinery@v0.0.0-20190221213512-86fb29eff628/pkg/util/runtime/runtime.go:51
/usr/local/go/src/runtime/panic.go:679
/usr/local/go/src/runtime/panic.go:199
/usr/local/go/src/runtime/signal_unix.go:394
/home/renuka/gopath/src/github.com/wso2/k8s-api-operator/api-operator/pkg/registry/amazon_ecr.go:98
/home/renuka/gopath/src/github.com/wso2/k8s-api-operator/api-operator/pkg/registry/registry.go:82
/home/renuka/gopath/src/github.com/wso2/k8s-api-operator/api-operator/pkg/controller/api/api_controller.go:1795
/home/renuka/gopath/src/github.com/wso2/k8s-api-operator/api-operator/pkg/controller/api/api_controller.go:598
/home/renuka/gopath/pkg/mod/sigs.k8s.io/controller-runtime@v0.1.10/pkg/internal/controller/controller.go:215
/home/renuka/gopath/pkg/mod/sigs.k8s.io/controller-runtime@v0.1.10/pkg/internal/controller/controller.go:158
/home/renuka/gopath/pkg/mod/k8s.io/apimachinery@v0.0.0-20190221213512-86fb29eff628/pkg/util/wait/wait.go:133
/home/renuka/gopath/pkg/mod/k8s.io/apimachinery@v0.0.0-20190221213512-86fb29eff628/pkg/util/wait/wait.go:134
/home/renuka/gopath/pkg/mod/k8s.io/apimachinery@v0.0.0-20190221213512-86fb29eff628/pkg/util/wait/wait.go:88
/usr/local/go/src/runtime/asm_amd64.s:1357
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xf2bf70]

goroutine 359 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/home/renuka/gopath/pkg/mod/k8s.io/apimachinery@v0.0.0-20190221213512-86fb29eff628/pkg/util/runtime/runtime.go:58 +0x105
panic(0x15d1a00, 0x264acf0)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/wso2/k8s-api-operator/api-operator/pkg/registry.glob..func1(0x2656080, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc00093c780, 0x45, 0xc00077306a, ...)
	/home/renuka/gopath/src/github.com/wso2/k8s-api-operator/api-operator/pkg/registry/amazon_ecr.go:98 +0x5b0
github.com/wso2/k8s-api-operator/api-operator/pkg/registry.IsImageExists(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1b24920, 0xc0004cb1a0, 0xc000d3c780, 0x1adafa0, ...)
	/home/renuka/gopath/src/github.com/wso2/k8s-api-operator/api-operator/pkg/registry/registry.go:82 +0x226
github.com/wso2/k8s-api-operator/api-operator/pkg/controller/api.isImageExist(0xc0003b4600, 0x17e9381, 0x1b, 0x17d3ca7, 0xb, 0xc, 0xc00077306a, 0x2)
	/home/renuka/gopath/src/github.com/wso2/k8s-api-operator/api-operator/pkg/controller/api/api_controller.go:1795 +0x2b6
github.com/wso2/k8s-api-operator/api-operator/pkg/controller/api.(*ReconcileAPI).Reconcile(0xc0003b4600, 0xc000a28ec9, 0x7, 0xc000a28eb0, 0xc, 0xc0006fbe80, 0xc0005a5cc0, 0x17b23a0, 0xc00084e000)
	/home/renuka/gopath/src/github.com/wso2/k8s-api-operator/api-operator/pkg/controller/api/api_controller.go:598 +0x1f6e
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc00016a960, 0xc0002c9800)
	/home/renuka/gopath/pkg/mod/sigs.k8s.io/controller-runtime@v0.1.10/pkg/internal/controller/controller.go:215 +0x20a
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1()
	/home/renuka/gopath/pkg/mod/sigs.k8s.io/controller-runtime@v0.1.10/pkg/internal/controller/controller.go:158 +0x36
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc000776530)
	/home/renuka/gopath/pkg/mod/k8s.io/apimachinery@v0.0.0-20190221213512-86fb29eff628/pkg/util/wait/wait.go:133 +0x5e
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000776530, 0x3b9aca00, 0x0, 0x1, 0xc0003c66c0)
	/home/renuka/gopath/pkg/mod/k8s.io/apimachinery@v0.0.0-20190221213512-86fb29eff628/pkg/util/wait/wait.go:134 +0xf8
k8s.io/apimachinery/pkg/util/wait.Until(0xc000776530, 0x3b9aca00, 0xc0003c66c0)
	/home/renuka/gopath/pkg/mod/k8s.io/apimachinery@v0.0.0-20190221213512-86fb29eff628/pkg/util/wait/wait.go:88 +0x4d
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start
	/home/renuka/gopath/pkg/mod/sigs.k8s.io/controller-runtime@v0.1.10/pkg/internal/controller/controller.go:157 +0x32e
```

**Steps to reproduce:**
- There is a untagged image in the ECR repository ( the repository that is supplied during the installation)
- Add new API

--- a/api-operator/pkg/registry/amazon_ecr.go
+++ b/api-operator/pkg/registry/amazon_ecr.go
@@ -94,8 +94,9 @@ var amazonEcr = &Config{
 
 		for _, id := range images.ImageIds {
 			// found the image with tag
-			logger.Info("Found the image tag from the AWS ECR repository", "RegistryId", awsRegistryId, "RepositoryName", awsRepoName, "image", imageName, "tag", tag)
-			if *id.ImageTag == fmt.Sprintf("%s-%s", imageName, tag) {
+			// untagged images 'id.ImageTag' returns nil; check nil before accessing the pointer
+			if id.ImageTag != nil && *id.ImageTag == fmt.Sprintf("%s-%s", imageName, tag) {
+				logger.Info("Found the image tag from the AWS ECR repository", "RegistryId", awsRegistryId, "RepositoryName", awsRepoName, "image", imageName, "tag", tag)
 				return true, nil
 			}
 		}

